### PR TITLE
Show an alert when the user saves an app or when the save failed

### PR DESCRIPTION
**Pull request by craicoverflow**
## Motivation

https://issues.jboss.org/browse/AEROGEAR-9045

## What

Added a Patternfly 4 Alert when an app is successfully saved or when the save failed. 

I wrapped the PF Alert in a custom component, as the PF Alert does not have any props available to control the visibility of the Alert.

## Why

To give the user visual feedback when the save has finished.

## Verification Steps
 
1. Go to the App Page.
2. Make changes to one of the versions.
3. Click `Save`.
4. You should see an Alert in the top right that reads "Save successful".
5. Click the `x` on the Alert. The Alert should now disappear.
6. Temporarily modify the [`UpdateAppVersions`](https://github.com/aerogear/mobile-security-service/blob/master/pkg/web/apps/apps_http_handler.go#L113) handler in the server to return a Bad Request.
7. Repeat steps **1-3**.
8. You should see an Alert in the top right that reads "Save failed".

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] Changes have been successfully verified by another team member 

## Progress

- [ ] Finished task
- [ ] TODO

### CODE REVIEW

**Comment by craicoverflow** [comment]
I have left the error message generic as I don't foresee any errors that can be rectified by the user (missing or invalid data). Thoughts?

File: ui/src/components/common/CustomAlert/CustomAlert.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      {...props}
+      action={<AlertActionCloseButton onClose={onClose} />}>
+      {children}
+    </Alert>
+  );
+};
+
+CustomAlert.propTypes = {
+  visible: PropTypes.bool.isRequired,
+  onClose: PropTypes.func,
```
Comment (davidffrench) [inline ui/src/components/common/CustomAlert/CustomAlert.js]: Do you think this should be a required proptype?

File: ui/src/components/common/CustomAlert/CustomAlert.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * @param {Object} props - The component props
+ */
+const CustomAlert = ({ visible, onClose, children, ...props }) => {
+  const [isVisible, setVisibility] = useState(visible);
+
+  useEffect(() => {
+    setVisibility(visible);
+  }, [visible]);
+
+  if (!isVisible) {
```
Comment (davidffrench) [inline ui/src/components/common/CustomAlert/CustomAlert.js]:
```suggestion
  if (! visible) {
```
Lines 15 - 20 do not look to be needed. You can switch the conditional to just use the suggestion above.

File: ui/src/containers/appView/AppPageContainer.js [hunk 3]
```diff
@@ -42,6 +47,45 @@ export class AppPageContainer extends React.Component {
...
+
+  /**
+   * Show an Alert
+   *
+   * @param {string} title - The alert title
+   * @param {*} variant - one of danger, success, warning, info
+   * @param {*} [description] - displays an optional Alert body
+   * @param {number} [timeout] - optionally set a duration of time before the alert disappears
+   */
+  showAlert (title, variant, description = null, timeout = 5000) {
```
Comment (davidffrench) [inline ui/src/containers/appView/AppPageContainer.js]: `timeout` is no longer required here.

File: ui/src/containers/appView/AppPageContainer.js [hunk 4]
```diff
@@ -42,6 +47,45 @@ export class AppPageContainer extends React.Component {
...
+    }
+  }
+
+  /**
+   * Show an Alert
+   *
+   * @param {string} title - The alert title
+   * @param {*} variant - one of danger, success, warning, info
+   * @param {*} [description] - displays an optional Alert body
+   * @param {number} [timeout] - optionally set a duration of time before the alert disappears
```
Comment (davidffrench) [inline ui/src/containers/appView/AppPageContainer.js]: `timeout` is no longer required here.

File: ui/src/containers/appView/AppPageContainer.js (hunk 3 above)
Comment (craicoverflow) [reply ui/src/containers/appView/AppPageContainer.js]: My bad, never pushed changes with it gone.

File: ui/src/containers/appView/AppPageContainer.js (hunk 4 above)
Comment (craicoverflow) [reply ui/src/containers/appView/AppPageContainer.js]: My bad, never pushed changes with it gone.

File: ui/src/components/common/CustomAlert/CustomAlert.js (hunk 1 above)
Comment (craicoverflow) [reply ui/src/components/common/CustomAlert/CustomAlert.js]: Yes it should be.

File: ui/src/components/common/CustomAlert/CustomAlert.js (hunk 2 above)
Comment (craicoverflow) [reply ui/src/components/common/CustomAlert/CustomAlert.js]: Yeah, I don't know why I thought I needed this :joy:

Comment (davidffrench) [reply ui/src/components/common/CustomAlert/CustomAlert.js]: was probably needed with the timeout logic.

**Pull request merged**